src/domManipulation.ts

```diff
@@ -4,16 +4,27 @@ import fetchWeatherForecast from "./api";
 
 // Get the parent element for the main cities/locations which holds all the buttons.
 const mainCities = document.getElementById("main-cities-nav");
+
 // Get all the button elements in this parent.
-const mainCityButtons = mainCities.getElementsByClassName("city-chip");
+let  mainCityButtons : HTMLCollectionOf<Element>;
+if (mainCities !== null) {
+  mainCityButtons = mainCities.getElementsByClassName("city-chip");
+}
+
 // Set the default city to the first option.
 let selectedMainCity = 0;
 
 // Toggle visibility of the pop-up
 const mainLocations = document.getElementById("main-cities-container");
 const customLocationPopup = document.getElementById("popup-container");
-mainLocations.style.visibility = "visible";
-customLocationPopup.style.visibility = "hidden";
+if (mainLocations !== null) {
```

Comment:
You could maybe use a guard clause here:
`if (mainLocations == null || customLocationPopup == null) { throw error; }
mainLocations.style.visibility = "visible";
customLocationPopup.style.visibility = "hidden";`
This is because I'm assuming that there is an error if those elements are null

Reply:
Updated guard clause:
```ts
if (!mainLocations || !customLocationPopup) {
  return;
}
```

Reply:
I removed these lines of code, since the only reason I needed to do this is because the popup's display was set to display by default, I have edited the style for the popup by hiding the display by default, making these lines redundant. I have edit the `closePopup()` function to match the given suggestions.